pkg/sql/logictest/logic.go

```diff
@@ -2058,6 +2101,153 @@ func (t *logicTest) processTestFile(path string, config testClusterConfig) error
 	return nil
 }
 
+func (t *logicTest) hasOpenTxns() bool {
+	for _, user := range t.clients {
+		if _, err := user.Exec("SET TRANSACTION PRIORITY NORMAL;"); !testutils.IsError(err, "there is no transaction in progress") {
```

Comment:
Could this end up changing the semantics of a test that has already tried to set the transaction priority to something other than normal?

Reply:
Good point, added a statement to reset it to the original txn priority.